integtest/src/main/java/org/corfudb/universe/cluster/docker/CorfuServerDockerized.java

```diff
@@ -0,0 +1,233 @@
+
+    /**
+     * Deploys a Corfu server / docker container
+     */
+    @Override
+    public CorfuServerDockerized deploy() throws NodeException {
+        log.info("Deploying the Corfu server. {}", params.getGenericName());
+
+        if(containerId.isPresent()){
+            throw new IllegalStateException("Corfu server already started. Parameters: " + params);
+        }
+
+        String id = deployContainer();
+
+        CorfuServerDockerized server = CorfuServerDockerized.builder()
+                .clusterParams(clusterParams)
+                .params(params)
+                .docker(docker)
+                .containerId(Optional.of(id))
+                .build();
+
+        server.addShutdownHook();
+        return server;
+    }
+
+    /**
+     * This method attempts to gracefully stops the Corfu server. After timeout, it will kill the Corfu server.
+     *
+     * @param timeout a duration after which the stop will kill the server
+     * @throws NodeException this exception will be thrown if the server can not be stopped.
+     */
+    @Override
+    public void stop(Duration timeout) throws NodeException {
+        log.info("Stopping the Corfu server. {}", params.getGenericName());
+
+        containerId.orElseThrow(() -> new NodeException("Can't STOP container. Invalid docker container id"));
+        containerId.ifPresent(id -> {
+            try {
+                docker.stopContainer(id, (int) timeout.getSeconds());
+            } catch (DockerException | InterruptedException e) {
+                throw new NodeException("Can't stop Corfu server", e);
+            }
+        });
+    }
+
+    /**
+     * Immediately kill the Corfu server.
+     *
+     * @throws NodeException this exception will be thrown if the server can not be killed.
+     */
+    @Override
+    public void kill() throws NodeException {
+        log.info("Killing the Corfu server. {}", params.getGenericName());
+
+        containerId.orElseThrow(() -> new NodeException("Can't KILL container. Invalid docker container id"));
+        containerId.ifPresent(id -> {
+            try {
+                docker.killContainer(id);
+            } catch (DockerException | InterruptedException ex) {
+                throw new NodeException("Can't kill Corfu server", ex);
+            }
+        });
+    }
+
+    /**
+     * Deploy and start docker container, expose ports, connect to a network
+     *
+     * @return docker container id
+     */
+    private String deployContainer() throws NodeException {
+        ContainerConfig containerConfig = buildContainerConfig();
+
+        String id;
+        try {
+            ContainerCreation creation = docker.createContainer(containerConfig, params.getGenericName());
+            id = creation.id();
+            docker.connectToNetwork(id, docker.inspectNetwork(clusterParams.getNetworkName()).id());
+
+            docker.startContainer(id);
+        } catch (InterruptedException | DockerException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)